### PR TITLE
Sample meta-data cleanup and minor re-org

### PR DESCRIPTION
**Pull request by nashif**
Also...
In sanitycheck, do not treat lpcxpresso54114_m0 a default test board, we do
have bbc microbit for that covering m0.

### CODE REVIEW

File: tests/posix/mutex/testcase.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 tests:
-  posix.mutex:
+  portability.posix:
```
Comment (andrewboie) [inline tests/posix/mutex/testcase.yaml]: why was this renamed?

**Review by tyrantlizard77 — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for docs
